### DISCUSSION TITLE
Can plugin provide search ability?

### DISCUSSION THREAD

**Discussion by linonetwo** [TiddlyWiki/TiddlyWiki5#5698] [Ideas]
I'd like to create a plugin that uses https://github.com/meilisearch to provide search results, is there an API to hook into the search form in TW?

**Comment by saqimtiaz** [comment]
@linonetwo the search results and form in TW are all implemented in wikitext.

The best way to go about this might be to implement a filter operator that uses meilisearch, and then add your own search results tiddler using this filter operator: https://tiddlywiki.com/#Customising%20search%20results

**Reply by linonetwo** [reply]
Thanks for suggestion, this will be a good way to test the integration. But I actually want to boost the native search form with typo-tolerance and tokenized CJK support.

**Reply by saqimtiaz** [reply]
@linonetwo there are a few options:

- You can create a new search results tiddler as above and set that as the default in `$:/config/SearchResults/Default`
- Or you could override the core shadow tiddler `$:/core/ui/DefaultSearchResultList` to use your new search filter operator
- Lastly you could override the core `search[]` operator, but I would not recommend this as it may be used by other plugins etc.

**Comment by pmario** [comment]
meilisearch is a "search server". So it will be more or less useless for single file wikis. 

There is the https://github.com/hoelzro/tw-full-text-search/ plugin, which creates a "local" index, per request or at startup. It does have a "fuzzy search" function, which isn't activated by default. ... So there would be a little adjustment needed. 

Fuzzy-search will be done if you type: `tipo~1` ... which imo should be part of the default search UI. ...
